flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/state/ConfigMapView.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.operator.autoscaler.state;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.util.Preconditions;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.dsl.Resource;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+class ConfigMapView {
+
+    enum State {
+        NEEDS_CREATE,
+        NEEDS_UPDATE,
+        UP_TO_DATE
+    }
+
+    private State state;
+
+    private ConfigMap configMap;
+
+    private final Function<ConfigMap, Resource<ConfigMap>> resourceRetriever;
+
+    public ConfigMapView(
+            Supplier<ConfigMap> configMapRetriever,
+            Supplier<ConfigMap> configMapBuilder,
+            Function<ConfigMap, Resource<ConfigMap>> resourceRetriever) {
+        var existingConfigMap = configMapRetriever.get();
+        if (existingConfigMap != null) {
+            refreshConfigMap(existingConfigMap);
+        } else {
+            this.configMap = configMapBuilder.get();
+            this.state = State.NEEDS_CREATE;
+        }
+        this.resourceRetriever = resourceRetriever;
```

Comment:
I think we could further simplify this by keeping only:
```
Resource<ConfigMap> resource:
ConfigMap configMap;
State state;
```


And pass those directly in the constructor / through a static create method instead of the Suppliers/Functions which feel a bit unnecessary

Reply:
The current state is the following btw:

```
Function<ConfigMap, Resource<ConfigMap>> resource:
ConfigMap configMap;
State state;
```

That occupies less space than what you are suggesting. Resource is only a pointer to an existing method.